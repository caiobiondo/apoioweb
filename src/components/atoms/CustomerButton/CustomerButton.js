import React from 'react';
import { FloatingActionButton, Icon } from 'natura-ui';
import { IconWrapper, Wrapper } from './CustomerButton.styles';

const OrderAddButton = props => {
  return (
    <Wrapper onClick={() => props.action()} remove={props.remove}>
      <FloatingActionButton iconWrapper={IconWrapper}>
        <Icon file="ico_times" />
      </FloatingActionButton>
    </Wrapper>
  );
};

export default OrderAddButton;
