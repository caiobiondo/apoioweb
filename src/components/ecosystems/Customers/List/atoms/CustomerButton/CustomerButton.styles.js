import styled from 'styled-components';
import { white } from 'styles/colors';

export const IconWrapperRemove = styled.div`
  svg {
    transform: rotate(0deg) !important;
  }
`;

export const Wrapper = styled.div`
  svg {
    fill: ${white};
    width: 30px;
    height: 30px;
    transition: all 0.3s ease;
  }

  ${props => {
    return props.remove ? IconWrapperRemove : null;
  }};
`;

export const IconWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;
