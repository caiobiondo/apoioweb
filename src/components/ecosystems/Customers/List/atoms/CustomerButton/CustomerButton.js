import React from 'react';
import { FloatingActionButton, Icon } from 'natura-ui';
import { IconWrapper, Wrapper } from './CustomerButton.styles';

const CustomerButton = props => {
  const icon = props.remove ? 'ico_trash' : 'ico_add_customer';
  return (
    <Wrapper onClick={props.action}>
      <FloatingActionButton iconWrapper={IconWrapper}>
        <Icon file={icon} />
      </FloatingActionButton>
    </Wrapper>
  );
};

export default CustomerButton;
