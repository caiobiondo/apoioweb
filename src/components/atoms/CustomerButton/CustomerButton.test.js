import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import CustomerButton from './CustomerButton';

describe('CustomerButton', () => {
  it('should render a floating button with add order icon', () => {
    const renderer = new ShallowRenderer();

    renderer.render(<CustomerButton />);
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });
});
