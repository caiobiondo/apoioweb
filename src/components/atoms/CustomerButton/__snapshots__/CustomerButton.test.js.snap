// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerButton should render a floating button with add order icon 1`] = `
<CustomerButtonstyles__Wrapper
  onClick={[Function]}
>
  <CustomFloatingActionButton
    iconWrapper={[Function]}
  >
    <Icon
      file="ico_add_customer"
    />
  </CustomFloatingActionButton>
</CustomerButtonstyles__Wrapper>
`;
